InvenTree/order/serializers.py

```diff
@@ -912,7 +912,7 @@ class Meta:
 class SalesOrderShipmentSerializer(InvenTreeModelSerializer):
     """Serializer for the SalesOrderShipment class."""
 
-    allocations = SalesOrderAllocationSerializer(many=True, read_only=True, location_detail=True)
+    allocations = SalesOrderAllocationSerializer(many=True, read_only=True, location_detail=True, order_detail=True)
```

Comment:
Suggestion: maybe make both location and order detail a flag to reduce unneeded traffic.